spec/compiler/type_inference/global_spec.cr

```diff
@@ -185,6 +185,20 @@ describe "Global inference" do
       )) { types["Foo"] }
   end
 
+  it "doesn't infer from new if generic" do
+    assert_error %(
+      class Foo(T)
+        def self.new
+          10
+        end
+      end
+
+      $x = Foo.new
+      $x
+      ),
+      "undefined global variable '$x'"
```

Comment:
If I define `self.new` over a generic type, I think it should be called with all the type arguments. And whatever we do in generic class should be done in non-generic class for consistency.

I don't see the need of class level `#new` in generic classes without arguments. But maybe I am missing something.